packages/appstore/src/components/wallet-cards-carousel/useCarousel.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState, useCallback, useEffect, useLayoutEffect } from 'react';
+
+export interface WalletCardsCarouselResult {
+    readonly pages: number[];
+    readonly activePageIndex: number;
+    readonly prev: () => void;
+    readonly next: () => void;
+    readonly goTo: (pageIndex: number) => void;
+    readonly scrollRef: (el: HTMLElement | null) => void;
+}
+
+interface WalletCardsCarouselState {
+    readonly pages: number[];
+    readonly activePageIndex: number;
+}
+
+export const useWalletCardCarousel = (): WalletCardsCarouselResult => {
+    const scrollDimension = 'scrollWidth';
+    const clientDimension = 'clientWidth';
+    const nearSidePos = 'left';
+    const scrollPos = 'scrollLeft';
+
+    const [scrollEl, setScrollEl] = useState<HTMLElement | null>(null);
```

Comment:
@sergei-deriv `useRef` 👀 
This is making a loop re-rendering as the state changes every time cause it's an element 😬

Let's just use a library instead of creating our own thing 🙏🏻